cns/restserver/util.go

```diff
@@ -671,22 +667,20 @@ func (service *HTTPRestService) SendNCSnapShotPeriodically(ncSnapshotIntervalInM
 	}
 }
 
-func (service *HTTPRestService) validateIpConfigRequest(ipConfigRequest cns.IPConfigRequest) (cns.KubernetesPodInfo, int, string) {
-	var podInfo cns.KubernetesPodInfo
-
+func (service *HTTPRestService) validateIpConfigRequest(ipConfigRequest cns.IPConfigRequest) (cns.PodInfo, int, string) {
 	if service.state.OrchestratorType != cns.KubernetesCRD {
-		return podInfo, UnsupportedOrchestratorType, fmt.Sprintf("ReleaseIPConfig API supported only for kubernetes orchestrator")
+		return nil, UnsupportedOrchestratorType, "ReleaseIPConfig API supported only for kubernetes orchestrator"
 	}
 
 	if ipConfigRequest.OrchestratorContext == nil {
-		return podInfo, EmptyOrchestratorContext, fmt.Sprintf("OrchastratorContext is not set in the req: %+v", ipConfigRequest)
+		return nil, EmptyOrchestratorContext, fmt.Sprintf("OrchastratorContext is not set in the req: %+v", ipConfigRequest)
 	}
 
 	// retrieve podinfo  from orchestrator context
-	if err := json.Unmarshal(ipConfigRequest.OrchestratorContext, &podInfo); err != nil {
+	podInfo, err := cns.NewPodInfoFromIPConfigRequest(ipConfigRequest)
```

Comment:
If the PodInfoVersion is set to InterfaceId, then can you check if PodInfo InterfaceId is not empty? May be do it inside Unmarshal wrapper